synapse/module_api/__init__.py

```diff
@@ -768,6 +801,11 @@ def generate_short_term_login_token(
                to get this token, if any. This is encoded in the token so that
                /login can report stats on number of successful logins by IdP.
         """
+        warnings.warn(
+            "ModuleApi.generate_short_term_login_token() is deprecated "
+            "in favor of ModuleApi.create_login_token().",
```

Comment:
I would suggest something like:

```suggestion
            "A module configured on this server uses ModuleApi.generate_short_term_login_token(), "
            "which is deprecated in favor of ModuleApi.create_login_token(), and will be removed in "
            "Synapse 1.xx.",
```

Since the person reading the message (the server's admin) is not necessarily the same as the one using the method.

Reply:
Done 👍